daemon/policy.go

```diff
@@ -451,10 +444,18 @@ func (d *Daemon) ReactToRuleUpdates(wg *sync.WaitGroup, epsToBumpRevision *polic
 		epp.PolicyRevisionBumpEvent(rev)
 	})
 
-	epsToRegen.Mutex.RLock()
 	// Regenerate all other endpoints.
-	endpointmanager.RegenerateEndpointSetSignalWhenEnqueued(d, &endpoint.ExternalRegenerationMetadata{Reason: "policy rules added"}, epsToRegen.IDs, &enqueueWaitGroup)
-	epsToRegen.Mutex.RUnlock()
+	regenMetadata := &endpoint.ExternalRegenerationMetadata{Reason: "policy rules added"}
+	epsToRegen.ForEach(&enqueueWaitGroup, func(ep policy.Endpoint) {
+		if ep != nil {
+			switch e := ep.(type) {
+			case *endpoint.Endpoint:
+				e.RegenerateSync(d, regenMetadata)
```

Comment:
Per [my earlier comment](https://github.com/cilium/cilium/pull/8340#issuecomment-503291202) this should be async and we should signal once we have enqueued a regeneration (e.g., not wait on result of (`ep.Regenerate`).